lib/create-file.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+var File = require('vinyl'),
+	fs = require('fs');
+
+/**
+ * Creates a vinyl file
+ * @function createFile
+ *
+ * @param {STRING}  filePath  relative path to file
+ * @param {STRING}  cwd  current working directory
+ * @param {STRING}  base  directory where we start looking for the file
+ * @param {STRING}  type  stream|null
+ */
+exports.createFile = function createFile (filePath, cwd, base, type) {
+  var contents;
+
+  if (type == 'stream') {
+    contents = fs.createReadStream(filePath);
+  } else {
+    contents = fs.readFileSync(filePath);
```

Comment:
what I'd really like to do is use promises instead. I'd specifically like to have a function within this utilities repo which does a readfile using promises and then use that for any async file reading. (I want one for writefile too)

So far, my attempts to set that up have failed. Promises still baffle me.

can you help @beynya ?